sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/jdbc/JDBCOptions.scala

```diff
@@ -36,4 +36,9 @@ private[jdbc] class JDBCOptions(
   val upperBound = parameters.getOrElse("upperBound", null)
```

Comment:
I see some of the existing code has a `getOrElse(..., sys.error...)`. I think we should switch to either

```
require(foo.isDefined, ...)
... foo.get(...)
```
or
```
foo.getOrElse(..., throw new IllegalArgumentException(...))
```
One problem is that `sys.error` just generates a bald `RuntimeException`.